### ISSUE TITLE
Unable to load asset: images/flag.png

### ISSUE THREAD

**Issue by Passaro1** [flutter/flutter#72882]
Good afternoon!

Issue:
======== Exception caught by image resource service ================================================

The following assertion was thrown resolving an image codec:

Unable to load asset: images/flag.png

When the exception was thrown, this was the stack: 

#0      PlatformAssetBundle.load (package:flutter/src/services/asset_bundle.dart:227:7)

<asynchronous suspension>

#1      AssetBundleImageProvider._loadAsync (package:flutter/src/painting/image_provider.dart:670:14)

<asynchronous suspension>

<asynchronous suspension>

Image provider: AssetImage(bundle: null, name: "images/flag.png")

________________________________________________
Code: 

class _QuizAppState extends State<QuizApp> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("True Citizen"),
        centerTitle: true,
        backgroundColor: Colors.blueGrey,
      ),
      body: Center(
        child: Image(
           image: AssetImage("images/flag.png"),height: 280,width: 280,
      ),
    ));
  }
}

_____________________________
 # To add assets to your application, add an assets section, like this:
assets:
     - assets/flag.png
     - assets/view.jpg
  #   - images/a_dot_ham.jpeg

Tried several way of making the flag appear but the result is always the same....any ideas or what I am missing?

Thank you!

**Comment by AbhishekDoshi26** [comment]
@Passaro1 AssetImage will not load your image on screen. To load the image, use Image.asset() and pass your image path inside it.

**Comment by Passaro1** [comment]
Thank you very much for the help...but now it seems something else is wrong..


![Error](https://user-images.githubusercontent.com/31216683/103029816-9540ea80-4528-11eb-91f1-b3ebb92bc493.png)

Hope we can keep this going...it will be very helpful!

Thanks!

**Comment by AbhishekDoshi26** [comment]
@Passaro1 you don't need to specify Image() you can directly use Image.asset as the child of Center ().

So basically it will be:
Center(
    child: Image.asset(),
),

**Comment by anirudh-jwala** [comment]
@Passaro1 Image widget take providers like `FileImage`, `MemoryImage`, `NetworkImage`, etc. Alternatively you can directly use `AssetImage` in place of entire Image() widget and mentioning of a provider.

**Comment by azbuky** [comment]
@Passaro1 you seem to have the flag image at `assets/images/flag.png` in your project

You should double check your pubspec.yaml, the assets section should look something like
```
  assets:
    - assets/images/flag.png
```

You should reference the flag image using `'assets/images/flag.png'` in your code.

More details about assets in the flutter docs - [Adding assets and images](https://flutter.dev/docs/development/ui/assets-and-images)

**Comment by Passaro1** [comment]
@AbhishekDoshi26 Thank you for the afford....but even with the changes you proposed it did not worked, I also double checked .yaml as proposed by @azbuky....I am having exact same "error"....here are the pictures.
![error1](https://user-images.githubusercontent.com/31216683/103102762-80795b00-45eb-11eb-9fb8-722ce0f6fedc.png)
![error1](https://user-images.githubusercontent.com/31216683/103102763-8111f180-45eb-11eb-87bd-5a8c31cc2616.png)

**Comment by Passaro1** [comment]
As you can see....the flag still does not appear and it brings same error.....I am really frustrated! But happy with all the help...

**Comment by AbhishekDoshi26** [comment]
@Passaro1 the alignment of assets in pubspec.yaml file is incorrect, as I can see in the screenshot. Make it correct, terminate the app, do pub get and then run again.

**Comment by Passaro1** [comment]
Hum....strange because I ran pub get and does not give me any error. It returns 0 every time....I tried several ways always returning 0 errora and still same thing...any other idea...On Dec 24, 2020 1:40 PM, Abhishek Doshi <notifications@github.com> wrote:
@Passaro1 the alignment of assets in pubspec.yaml file is incorrect, as I can see in the screenshot. Make it correct, terminate the app, do pub get and then run again.

**Comment by Passaro1** [comment]
Man...you were correct! @AbhishekDoshi26 Man this is isane...hahahha...You have no idea how many combinations I tried! And I learned with all this....My get pub showed no "error" but still was wrong...Thank you for all help guys! I will post the result here.

**Comment by Passaro1** [comment]
![Correct Indent pubspec yaml](https://user-images.githubusercontent.com/31216683/103156788-7a65b480-477a-11eb-8f6c-522c4420c4c9.png)

**Comment by Passaro1** [comment]
💯 !

**Comment by AbhishekDoshi26** [comment]
Cool! It's good that your problem got solved and you learnt 🎉
@anirudh-jwala

**Comment by Algy-Jr12** [comment]
If anyone tried all the above and didn't succeed, try quitting (terminate the application on the device) and run it again (flutter run), this should update (add) all new assets added on pubspec.yaml file.

I was having the same error, I tried all the above methods, but it wasn't working, and when I decided to quit the application on debug mode and rerun the command "flutter run" and Xcode built it (Compile, link, and sign), this updated (add) all new assets from pubspec.yaml file.

PS: I was using VS Code, Xcode Simulator for debugging.
